develop/smart-contracts/evm-toolkit/dev-environments/remix.md

```diff
@@ -0,0 +1,107 @@
+
+The interface is organized into three main sections: the Extensions panel on the right, the Code Editor in the center, and the File Explorer on the left.
+
+## Creating a New Contract
+
+To create a new contract using the Polkadot Remix IDE, you can follow these steps:
+
+1. Select the **Create a new file** button in the `contracts` folder
+
+    ![](/images/develop/smart-contracts/evm-toolkit/dev-environments/remix/remix-2.webp)
+
+2. Name your file with a `.sol` extension, in this case, `Counter.sol`
+
+    ![](/images/develop/smart-contracts/evm-toolkit/dev-environments/remix/remix-3.webp)
+
+3. Write your Solidity code in the editor
+
+    ![](/images/develop/smart-contracts/evm-toolkit/dev-environments/remix/remix-4.webp)
+
+    The code of the contract above is the following:
+
+    ???- "Counter.sol"
+        
+        ```solidity
+        --8<-- 'code/develop/smart-contracts/evm-toolkit/dev-environments/remix/Counter.sol'
+        ```
+
+## Compiling Your Contract
+
+1. To compile your contract, you need to
+
+    1. Navigate to the **Solidity Compiler** tab (third icon in the left sidebar)
+    2. Select**Compile** or use `Ctrl+S`
+
+        ![](/images/develop/smart-contracts/evm-toolkit/dev-environments/remix/remix-5.webp)
+    
+        !!! note
+            Compilation errors and warnings appear in the terminal panel at the bottom of the screen.
+
+After compiling your contract, you can navigate to the **File Explorer** tab (first icon in the left sidebar) and check that the `artifact` folder exists and that the `Counter_metadata.json` and `Counter.json` files have also been created.
+
+![](/images/develop/smart-contracts/evm-toolkit/dev-environments/remix/remix-6.webp)
+
+## Deploying Contracts
+
+1. To deploy your contract, you need to
+
+    1. Navigate to the **Deploy & Run Transactions** tab (fourth icon in the left sidebar)
+    2. Select your deployment environment, in this case, **Westend Testnet - MetaMask**
+    3. Select the **Deploy and Confirm** button
+
+        ![](/images/develop/smart-contracts/evm-toolkit/dev-environments/remix/remix-7.webp)
+
+        Once your contract is deployed successfully, you will see the following output in the REMIX terminal:
+
+        ![](/images/develop/smart-contracts/evm-toolkit/dev-environments/remix/remix-8.webp)
+
+## Interacting with Contracts
+
+Once deployed, your contract appears in the **Deployed/Unpinned Contracts** section:
+
+1. Expand the contract to view available methods
+
+    ![](/images/develop/smart-contracts/evm-toolkit/dev-environments/remix/remix-9.webp)
+
+    !!! tip
+        Pin your frequently used contracts to the **Pinned Contracts** section for easy access.
+
+2. To interact with the contract, you can select any of the exposed methods
+
+    ![](/images/develop/smart-contracts/evm-toolkit/dev-environments/remix/remix-10.webp)
+
+    In this way, you can interact with your deployed contract by reading its state or writing to it.
```

Comment:
You can also add that the color of the button represents the kind of interaction you can do:
red - modify state + payable
orange - modify state
blue - read state